File: JAVA/2465. Number of Distinct Averages.java
class Solution {
    public int distinctAverages(int[] n) {
        HashSet<Integer>s=new HashSet<>();
        Arrays.sort(n);
        for(int i=0;i<n.length/2;i++){
            s.add(n[i]+n[n.length-1-i]);
        }
        return s.size();
    }
}

File: C++/2465. Number of Distinct Averages.cpp
class Solution {
public:
    int distinctAverages(vector<int>& n) {
        set<int>s;
        sort(n.begin(),n.end());
        for(int i=0;i<n.size()/2;i++){
            s.insert(n[i]+n[n.size()-1-i]);
        }
        return s.size();
    }
};